### PR TITLE
Tooling docs updates (versioning linked to `roachtest`, title updates, etc)

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes https://github.com/cockroachdb/docs/issues/9815.
Fixes https://github.com/cockroachdb/docs/issues/8796.

This PR remote-includes the version numbers from the available driver/ORM test files in [`roachtest`](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/roachtest). This will make it much easier for us to scale support, while maintaining accuracy in our recommendations without too much docs/engineering overhead.

The diffs on the PR will look fairly meaningless - you have to look at the preview.

Note that I omitted the following, as their version formatting couldn't be directly included from the `roachtest` files:

- https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/psycopg.go
- https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/sqlalchemy.go
- https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/activerecord.go (less of a formatting thing and more just bc we support a lower, commonly-used version that is being tested atm)

If we want to tie these versions to the docs, the formatting should be updated at the source.

Purely from a docs perspective, I would very much prefer that all of these versions be kept in a separate file, in `roachtest`, and then read into each individual file at runtime or something.

### CODE REVIEW

**Review by ericharmeling — COMMENTED** [review]
TFTR @rafiss !

I think I got to all your comments, but I did have a quick question re: the django versions. PTAL.

> out of curiosity, what happens if we change a filename or variable name being referenced?

I think it just returns an empty string or an error message in the place of the included text. It doesn't look like these variables get updated super often, but we should make sure that engineers are aware of the docs dependency. Perhaps we can update the `roachtest` README to warn about the docs dependency. (Or, even better, add some build test against this page into the main test file or something). Thoughts?

-----

Also... I decided to make some other updates and sneak them into the latest commit as well. These updates are more about structure/docs-y than anything:

- Improvements to page titles.
- The addition of a "Tutorial" column to each of the tables. (We should really just have a single link to the external site for the tool, and then have a separate, explicit link to any tutorials that we want to link to.) 
- Copied the CRDB vs community support language over to 20.2
 
LMK if you see any issues with these updates - they will be writer-reviewed in any case.

File: _includes/v20.2/misc/tooling.md
Comment (rafiss) [reply _includes/v20.2/misc/tooling.md]: > var djangoSupportedTag = "cockroach-3.2.x"
> var djangoCockroachDBSupportedTag = "3.2.1"

is correct -- not super obvious but `cockroach-3.2.x` is just a branch of django that has testing changes to work with cockroachdb. `3.2.1` is the version of our adapter.

actually, we should always be recommending the latest versions of all our adapters (sqlalchemy-cockroachdb, django-cockroachdb, sequelize-cockroachdb, activerecord-cockroachdb-adapter, hibernate cockroach dialect). it would be more important to make sure the ORM version we list here is fixed.

maybe for those it's easier to just link to the github/pip/rubygem/npmjs pages?

File: _includes/v20.2/misc/tooling.md
Comment (rafiss) [reply _includes/v20.2/misc/tooling.md]: oh sorry, i wasn't clear. we support activerecord 5.2, 6.0, and 6.1 -- the major.minor version of our adapter is compatible with the respective version of activerecord.

**Comment by vy-ton** [comment]
> Note that I omitted the following, as their version formatting couldn't be directly included from the roachtest files:
https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/activerecord.go (less of a formatting thing and more just bc we support a lower, commonly-used version that is being tested atm)

This shouldn't be true. We support the latest `6.1` version. For tooling where we support multiple versions, we could just mention the latest version. Older versions are available via the package manager.

+1 to pulling the documented version from `roachtest`. @rafiss it does mean we will want to be more aggressive with pulling in new adapter versions (for example Django 3.2 adapter is available)

**Review by vy-ton — APPROVED** [review]

**Comment by ericharmeling** [comment]
> This shouldn't be true. We support the latest 6.1 version. For tooling where we support multiple versions, we could just mention the latest version. Older versions are available via the package manager.

@vy-ton 

So in the case of ActiveRecord you're suggesting we just recommend 6.1, even though a lot of people still probably use 5.x? (I suppose we are doing this in the case of Django anyways...)

**Comment by vy-ton** [comment]
> > This shouldn't be true. We support the latest 6.1 version. For tooling where we support multiple versions, we could just mention the latest version. Older versions are available via the package manager.
> 
> @vy-ton
> 
> So in the case of ActiveRecord you're suggesting we just recommend 6.1, even though a lot of people still probably use 5.x? (I suppose we are doing this in the case of Django anyways...)

Maybe we should make our lives easier and remove the specific adapter version altogether. The links direct users to the right package manager, where you can see the latest and older versions

**Comment by ericharmeling** [comment]
> Maybe we should make our lives easier and remove the specific adapter version altogether. The links direct users to the right package manager, where you can see the latest and older versions

Hmm... Just for the ActiveRecord adapter, or for everything? 

+1 for removing the specific ActiveRecord adapter, but I do think having a specific version listed (i.e., the latest tested version) provides some amount of value.

Here's an idea: We add a note, encouraging users to try the latest version for all supported tools, and then to change the wording on the tables from "Recommend version" -> "Latest version tested" or something like that.

@vy-ton @rafiss Thoughts?

**Comment by rafiss** [comment]
We are the maintainers of the adapters, so the latest version tested is always going to be the latest adapter version.

So I think to Vy's point, we can link to:
- https://pypi.org/project/django-cockroachdb/
- https://pypi.org/project/sqlalchemy-cockroachdb/
- https://rubygems.org/gems/activerecord-cockroachdb-adapter/
- https://www.npmjs.com/package/sequelize-cockroachdb

But we maybe DO want to make sure we specify the version of the ORM itself that has been tested... that info can either be on this page, or just leave it to the README of the adapter we link to.

**Comment by vy-ton** [comment]
For ORMs with an adapter, we list 2 versions - the `ORM version` and the `adapter package version`.

To Rafi's point, I would like to see
* the ORM version that has been tested
* link to adapter package - This is where I think we could omit any version. If it's easier, I am also good with listing the version that is tested. 
* +1 to `Recommended version` --> `Latest version tested`

**Comment by ericharmeling** [comment]
Got it. I'll update the PR accordingly. Thank you both!

**Comment by vy-ton** [comment]
@ericharmeling fyi, you can bump the Javascript driver and Sequelize support to `Full`

**Comment by ericharmeling** [comment]
> fyi, you can bump the Javascript driver and Sequelize support to `Full`

Done.

**Review by mikeCRL — APPROVED** [review]
LGTM. One note.

**Review by ericharmeling — COMMENTED** [review]
TFTR, @mikecrl 

I replied, asking @vy-ton for her opinion on the matter.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10756#-:-MdScEF604qD-0o4zec0:b-9o81iz)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling and @vy-ton)

___
*[_includes/v21.1/misc/tooling.md, line 8 at r2](https://reviewable.io/reviews/cockroachdb/docs/10756#-MbhaoxNtLsw4GsDBCkL-r2-8:-MdSZFbYC_Fzzb-542DV:bqy7opt) ([raw file](https://github.com/cockroachdb/docs/blob/8040957b67a973f26fec887929febf6521f25b34/_includes/v21.1/misc/tooling.md#L8)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

Should we be clearer that you should attempt to use more recent versions, to ensure you've received the latest features/fixes from the applicable tool, but to take care e.g. test the tool on non-production workloads and maintain backups? (As obvious as that may seem.) Or some other more nuanced note? I am not familiar enough with the tools to know the risks.
</blockquote></details>

That's a good point.

Here are my thoughts:

- I'm wary of making it seem too risky to use later versions of a tool, as we are committed to having full support for most of these tools. If a minor version update of any one tool requires additional non-production testing/backups due to our own uncertainty about how it will affect CRDB's compatibility with the tool, it might send the wrong message about our product's stability, our testing process, and our claims for full support for a tool.

- If we tell users to exercise caution when using the latest version of a tool, then I don't think we can recommend that they use the latest version of the tool in the same sentence. 

So here are some options:

1. We remove the recommendation, and add a simple note of caution instead.
2. We remove the recommendation, and add a more nuanced note about non-production testing and additional backups.
3. We keep the recommendation as-is, and trust that our testing covers as many cases as it can (and updates the version tested as often as possible), and then rely on users to open an issue if they run into any issues that we didn't catch. 
4. We remove the first "We recommend..." sentence from the paragraph, and just keep the second "If you encounter problems..." sentence.

@vy-ton  

Thoughts?

**Review by vy-ton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10756)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling and @mikeCRL)

___
*[_includes/v21.1/misc/tooling.md, line 8 at r2](https://reviewable.io/reviews/cockroachdb/docs/10756#-MbhaoxNtLsw4GsDBCkL-r2-8:-MdSmm5iDWND_aBajV9d:b-n4oit4) ([raw file](https://github.com/cockroachdb/docs/blob/8040957b67a973f26fec887929febf6521f25b34/_includes/v21.1/misc/tooling.md#L8)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

That's a good point.

Here are my thoughts:

- I'm wary of making it seem too risky to use later versions of a tool, as we are committed to having full support for most of these tools. If a minor version update of any one tool requires additional non-production testing/backups due to our own uncertainty about how it will affect CRDB's compatibility with the tool, it might send the wrong message about our product's stability, our testing process, and our claims for full support for a tool.

- If we tell users to exercise caution when using the latest version of a tool, then I don't think we can recommend that they use the latest version of the tool in the same sentence. 

So here are some options:

1. We remove the recommendation, and add a simple note of caution instead.
2. We remove the recommendation, and add a more nuanced note about non-production testing and additional backups.
3. We keep the recommendation as-is, and trust that our testing covers as many cases as it can (and updates the version tested as often as possible), and then rely on users to open an issue if they run into any issues that we didn't catch. 
4. We remove the first "We recommend..." sentence from the paragraph, and just keep the second "If you encounter problems..." sentence.

@vy-ton  

Thoughts?
</blockquote></details>

I like Option 4.

**Review by ericharmeling — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10756)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling and @mikeCRL)

___
*[_includes/v21.1/misc/tooling.md, line 8 at r2](https://reviewable.io/reviews/cockroachdb/docs/10756#-MbhaoxNtLsw4GsDBCkL-r2-8:-MdT4Rq7B-VoY0aMFaHZ:b-pimrx3) ([raw file](https://github.com/cockroachdb/docs/blob/8040957b67a973f26fec887929febf6521f25b34/_includes/v21.1/misc/tooling.md#L8)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

I like Option 4. 
</blockquote></details>

Okay - I went with option 4 for now. This option (and option 3) has the benefit of keeping our support levels stable in description. Throughout our docs, our "experimental" features have non-production testing recommendations and warnings - and IMO we don't want to use similar language for fully-supported tools (even if it's just a versioning thing).

**Pull request merged**
